src/components/ui/DownloadButton.tsx

```diff
@@ -0,0 +1,65 @@
+'use client';
+
+import { useState } from 'react';
+
+import { cva } from 'class-variance-authority';
+import { ArrowDownToLine } from 'lucide-react';
+
+const buttonStyles = cva(
+  'py-4 px-4 pr-6 rounded-full font-bold text-lg transition-all duration-300 cursor-pointer flex items-center justify-center',
+  {
+    variants: {
+      variant: {
+        outline: 'bg-transparent',
+      },
+      colorScheme: {
+        blue: 'text-blue border-2 border-blue hover:bg-blue hover:text-white',
+        white:
+          'text-white border-2 border-white hover:bg-white hover:text-blue',
+      },
+    },
+    defaultVariants: {
+      variant: 'outline',
+      colorScheme: 'blue',
+    },
+  }
+);
+
+export default function DownloadButton() {
+  const [, setIsClicked] = useState(false);
+
+  const handleDownload = async () => {
+    setIsClicked(true);
+
+    try {
+      const response = await fetch('/charts/SAMAHANOrganizationalChart.pdf');
+      const blob = await response.blob();
+      const url = window.URL.createObjectURL(blob);
+      const link = document.createElement('a');
+      link.href = url;
+      link.download = 'SAMAHAN Organizational Chart.pdf';
+      document.body.appendChild(link);
+      link.click();
+      document.body.removeChild(link);
+    } catch (error) {
+      console.error('Download failed:', error);
+    }
```

Comment:
This can be done in a simpler way without all of this logic. In your button component
```
<Link href="/charts/SAMAHANOrganizationalChart.pdf" download="SamahanOrganizationalChart">
            <Button>
              Download Chart
            </Button>
          </Link>
```